demisto_sdk/commands/download/downloader.py

```diff
@@ -439,305 +460,397 @@ def build_req_params(self):
-                "\nThe following files are available to be downloaded from Demisto instance:\n"
-            )
-            logger.info(tabulate(list_files, headers=["FILE NAME", "FILE TYPE"]))
-            return True
-        return False
+        Return a printable list of all custom content that's available to download from the configured XSOAR instance.
 
-    def handle_all_custom_content_flag(self) -> None:
-        """
-        Handles the case where the all custom content flag is given
-        :return: None
-        """
-        if self.all_custom_content:
-            custom_content_objects: list = self.get_custom_content_objects()
-            names_list: list = [cco["name"] for cco in custom_content_objects]
-            # Remove duplicated names, for example: IncidentType & Layout with the same name.
-            self.input_files = list(set(names_list))
+        Args:
+            custom_content_objects (dict[str, dict]): A dictionary mapping custom content's file names to objects.
 
-    def handle_regex_flag(self) -> None:
-        """
-        Handles the case where the regex flag is given
-        :return: None
+        Returns:
+            str: A printable list of all custom content that's available to download from the configured XSOAR instance.
         """
-        input_files_regex_match = []
-        if self.regex:
-            custom_content_objects: list = self.get_custom_content_objects()
-            names_list: list = [cco["name"] for cco in custom_content_objects]
+        tabulate_data: list[list[str]] = []
+
+        for file_name, file_object in custom_content_objects.items():
+            if item_name := file_object.get("name"):
+                file_type: FileType = file_object["type"]
+                tabulate_data.append([item_name, file_type.value])
 
-            for input_file in names_list:
-                if re.search(self.regex, input_file):
-                    input_files_regex_match.append(input_file)
-            self.input_files = input_files_regex_match
+        return tabulate(tabulate_data, headers=["CONTENT NAME", "CONTENT TYPE"])
 
-    def handle_init_flag(self) -> None:
+    def initialize_output_path(self, root_folder: Path) -> Path:
         """
-        Handles the case where the init flag is given
-        :return: None
+        Initialize output path with pack structure.
+
+        Args:
+            root_folder (Path): The root folder to initialize the pack structure in.
+
+        Returns:
+            Path: Path to the initialized output path.
         """
-        if not self.init:
-            return
+        logger.info("Initiating pack structure...")
 
-        root_folder = Path(self.output_pack_path)
         if root_folder.name != "Packs":
             root_folder = root_folder / "Packs"
+
             try:
                 root_folder.mkdir(exist_ok=True)
+
             except FileNotFoundError as e:
                 e.filename = str(Path(e.filename).parent)
```

Comment:
This is a function that creates a new directory (init flag), so the parent would be the actual folder we're trying to initialize the pack structure in. The "child" is obviously not found since `root_folder.mkdir(exist_ok=True)` was supposed to create it, but failed due to the parent not existing.